src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp

```diff
@@ -101,6 +118,134 @@ class PolicyManager final : public Singleton<PolicyManager>
             });
     }
 
+    void validateConfiguration()
+    {
+        for (auto it = m_configuration.begin(); it != m_configuration.end(); it++)
+        {
+            if (it.key().compare("vulnerabilty-detection"))
+            {
+                validateAndConfigureVulnerabilityDetection(*it);
+            }
+            else if (it.key().compare("indexer"))
+            {
+                validateAndConfigureIndexer(*it);
+            }
+            else
+            {
+                throw std::runtime_error("Invalid JSON object.");
+            }
+        }
+    }
+
+    /**
+     * @brief Validates and configures the vulnerability detection based on the provided JSON object.
+     *
+     * This function takes a JSON object as input, which is expected to contain configuration
+     * information for an vulnerability detection. It validates the JSON object to ensure it contains the
+     * required fields and has valid values.
+     * If the validation passes, no exception is throw.
+     *
+     * @param vdObj A constant reference to a JSON object representing the vulnerability detection's configuration.
+     *
+     * @note This function assumes that the provided JSON object follows a specific format.
+     * @note If validation fails, this function throw exceptions.
+     *
+     */
+
+    void validateAndConfigureVulnerabilityDetection(const nlohmann::json& vdObj)
```

Comment:
What do you think of sending `vdObj["vulnerability-detection"]` directly? It might simplify the method

Reply:
Done thx guys!